### PR TITLE
Fix calendar emails to be outlook compatible

### LINKED ISSUE #12885: Calendar Invites - Outlook integration
### Steps to reproduce
1. Create a Calendar Invite in Thunderbird - invite someone who's using Outlook 2010. 
2. Create a similar Calendar Invite in Nextcloud - invite same person using Outlook 2010

### Expected behaviour
I expect the same behaviour as with Outlook 2016, i.e. the invite gives me buttons on top of the invite and show me the calendar of Outlook/Exchange. However, with the thunderbird invite and invites from google, they are shown; but not with the invites from nextcloud. Thunderbird/nextcloud invites come from the same domain.

image

### Actual behaviour

No buttons are shown. 
image

### Server configuration

**Operating system**: Ubuntu 18

**Web server:** Apache2

**Database:** PostgreSQL 9.5

**PHP version:** 7.2

**Nextcloud version:** (see Nextcloud admin page) 14.x

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated

**Where did you install Nextcloud from:** Web

### PR DESCRIPTION
**Pull request by ChristophWurst**
* Resolves: https://github.com/nextcloud/server/issues/12885

## Summary

Resubmit and formatted version of https://github.com/nextcloud/server/pull/34324

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by tcitworld — APPROVED** [review]

File: apps/dav/lib/CalDAV/Schedule/IMipPlugin.php [hunk 1]
```diff
@@ -268,12 +268,18 @@ public function schedule(Message $iTipMessage) {
...
+		** We choose to work like Thunderbird Lightning.
+		** using a plain text text/calendar ics.
+		** This plain text text/calendar part is needed for
+		** Microsoft Outlook versions <= 2010 to work.
+		*/
+		$itip_msg = $iTipMessage->message->serialize();
+		$message->attachInline(
+				$itip_msg,
+				'text/calendar; method=' . $iTipMessage->method,
+				'UTF-8'
```
Comment (miaulalala) [inline apps/dav/lib/CalDAV/Schedule/IMipPlugin.php:281]: Will this still show the ics as a downloadable attachment? I guess it depends on each client implementation. Could we maybe do both? Attach inline and as a separate file?

Comment (ChristophWurst) [reply apps/dav/lib/CalDAV/Schedule/IMipPlugin.php:281]: Then please take over the PR. These are not my changes.

Comment (tcitworld) [reply apps/dav/lib/CalDAV/Schedule/IMipPlugin.php:281]: IIRC Thunderbird doesn't have any issue with that.

**Comment by ChristophWurst** [comment]
@miaulalala would this be realistic to finish for 26?

**Comment by miaulalala** [comment]
Questionable if symfony supports this: https://github.com/symfony/symfony/issues/44557

I will look into it tomorrow

**Comment by miaulalala** [comment]
The attachment is embedded now, but I have no outlook client to test this. @merlinwoff  can you please try?

**Comment by ChristophWurst** [comment]
The rebase went sideways

File: lib/public/Mail/IMessage.php [hunk 2]
```diff
@@ -39,6 +39,17 @@ interface IMessage {
 	 */
 	public function attach(IAttachment $attachment): IMessage;
 
+	/**
+	 * Can be used to "attach content inline" as message parts with specific MIME type and encoding.
+	 *
+	 * @param $body: body of the MIME part
+	 * @param $content-type: MIME Content-Type (e.g. text/plain or text/calendar)
+	 * @param $charset: Character Set (e.g. UTF-8)
```
Comment (ChristophWurst) [inline lib/public/Mail/IMessage.php:47]: `@param type name description`, e.g. `@param string $body body of the MIME part`

File: lib/public/Mail/IMessage.php [hunk 3]
```diff
@@ -42,9 +42,10 @@ public function attach(IAttachment $attachment): IMessage;
 	/**
 	 * Can be used to "attach content inline" as message parts with specific MIME type and encoding.
 	 *
-	 * @param $body: body of the MIME part
-	 * @param $content-type: MIME Content-Type (e.g. text/plain or text/calendar)
-	 * @param $charset: Character Set (e.g. UTF-8)
+	 * @param string $body  body of the MIME part
```
Comment (ChristophWurst) [inline lib/public/Mail/IMessage.php:45]:
```suggestion
	 * @param string $body body of the MIME part
```

**Review by ChristophWurst — COMMENTED** [review]
:smiling_face_with_tear:

**Review by miaulalala — APPROVED** [review]

File: lib/public/Mail/IMessage.php [hunk 4]
```diff
@@ -39,6 +39,18 @@ interface IMessage {
...
 
+	/**
+	 * Can be used to "attach content inline" as message parts with specific MIME type and encoding.
+	 *
+	 * @param string $body body of the MIME part
+	 * @param string $name the file name
+	 * @param string|null $contentType MIME Content-Type (e.g. text/plain or text/calendar)
+	 *
+	 * @return IMessage
+	 * @since 26.0.0
```
Comment (nickvergessen) [inline lib/public/Mail/IMessage.php:50]: 27

File: lib/public/Mail/IMessage.php [hunk 5]
```diff
@@ -39,6 +39,18 @@ interface IMessage {
...
+	 */
+	public function attachInline(string $body, string $name, string $contentType = null): IMessage;
```
Comment (nickvergessen) [inline lib/public/Mail/IMessage.php:52]: `?string $contentType = null`

**Review by nickvergessen — CHANGES_REQUESTED** [review]
Please mention in the 27 changes issue

**Comment by DaphneMuller** [comment]
Hello!
Thank you for your work on this! I noticed that your PR still has the label 'documentation'. I'm not sure if you already had a look at the documentation for your PR, but at Nextcloud we strive to document changes that affect other app developers or other admins before the release takes place, and I'm pinging you with the hope to get this done. 

Are you familiar with our documentation process already? 
Changes that affect app developers should be documented here:
https://docs.nextcloud.com/server/27/developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_27.html

Changes that affect administrators should be documented here:
https://docs.nextcloud.com/server/latest/admin_manual/release_notes/upgrade_to_27.html

While I understand sometimes it's handy to merge fast to get your changes in, it would be great if next time you could try to add the documentation before merging. You can find more information on that here:
https://docs.nextcloud.com/server/27/developer_manual/prologue/compatibility_app_ecosystem.html#documentation-procedures-of-changes-that-affect-app-developers

If all your documentation efforts are done, please remove the label 'documentation' and check the checkbox in your opening note and I'll stop bugging you :) 

Many thanks in advance and thanks again for your work!

**Pull request merged**